File: code/bit_manipulation/src/xor_swap/xor_swap.py
# Part of Cosmos by OpenGenus Foundation

# Swaps two given numbers making use of xor

# Works for both python 2 and python 3


def xorswap(n, m):
    n = m ^ n
    m = n ^ m
    n = m ^ n
    return n, m


n = 10
m = 15
print("Earlier A was equal to ", n, " and B was equal to ", m)
n, m = xorswap(n, m)
print("Now A is equal to ", n, " and B is equal to ", m)

File: code/bit_manipulation/src/xor_swap/xor_swap.js
// Part of Cosmos by OpenGenus Foundation

function xorSwap(a: number, b: number): void {
    console.log(`Before swap: a = ${a}, b = ${b}`);
    
    a = a ^ b; // Step 1: a now contains the XOR of both
    b = a ^ b; // Step 2: b gets the original value of a
    a = a ^ b; // Step 3: a gets the original value of b

    console.log(`After swap: a = ${a}, b = ${b}`);
}

// Example usage:
let x: number = 5;
let y: number = 10;
xorSwap(x, y);
